### PR TITLE
Tello3 detection

### PR DESCRIPTION
**Pull request by Skwarson96**

### CODE REVIEW

File: src/inference/hand_detection.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+from yolo import YOLO
+
+
+class HandDetector:
+
+    def __init__(self):
+        self.yolo = YOLO("models/cross-hands-tiny.cfg", "models/cross-hands-tiny.weights", ["hand"])
+        self.yolo.size = int(416)
+        self.yolo.confidence = float(0.2)
```
Comment (arekmula) [inline src/inference/hand_detection.py]: Add yolo size and confidence as parameter to class initialization and set their default values to 416 and 0.2

File: src/inference/hand_detection.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import tensorflow as tf
+import numpy as np
+
+from yolo import YOLO
+
+
+class HandDetector:
+
+    def __init__(self):
+        self.yolo = YOLO("models/cross-hands-tiny.cfg", "models/cross-hands-tiny.weights", ["hand"])
```
Comment (arekmula) [inline src/inference/hand_detection.py:11]: Add information to README.md to put model in that directory

File: src/inference/hand_detection.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import cv2
+import sys
+import tensorflow as tf
```
Comment (arekmula) [inline src/inference/hand_detection.py]: import not used

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
 numpy~=1.20.2
 setuptools~=56.0.0
-h264decoder~=0.0.0
-opencv-python~=4.5.1.48
\ No newline at end of file
+opencv-python~=4.5.1.48
+tensorflow~=2.4.1
```
Comment (arekmula) [inline requirements.txt]: requirement not used

File: src/inference/hand_detection.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        conf_sum = 0
+        detection_count = 0
+
+        boxes = []
+
+        for detection in results:
+            id, name, confidence, x, y, w, h = detection
+            boxes.append((x, y, w, h))
+
+            if draw_bb:
```
Comment (arekmula) [inline src/inference/hand_detection.py]: draw_bb is not defined in this scope, so I guess this script will not even work :p

File: src/inference/hand_detection.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                conf_sum += confidence
+                detection_count += 1
+
+                # draw a bounding box rectangle and label on the image
+                color = (0, 0, 255)
+                cv2.rectangle(img_resize, (x, y), (x + w, y + h), color, 3)
+                text = "%s (%s)" % (name, round(confidence, 2))
```
Comment (arekmula) [inline src/inference/hand_detection.py]: Use Python3 fstrings

File: src/inference/hand_detection.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class HandDetector:
+
+    def __init__(self):
+        self.yolo = YOLO("models/cross-hands-tiny.cfg", "models/cross-hands-tiny.weights", ["hand"])
+        self.yolo.size = int(416)
+        self.yolo.confidence = float(0.2)
+
+    def predict(self, img, draw):
```
Comment (arekmula) [inline src/inference/hand_detection.py]: change to should_draw_results instead of draw

File: src/inference/hand_detection.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                color = (0, 0, 255)
+                cv2.rectangle(img_resize, (x, y), (x + w, y + h), color, 3)
+                text = "%s (%s)" % (name, round(confidence, 2))
+                cv2.putText(img_resize, text, (x, y - 5), cv2.FONT_HERSHEY_SIMPLEX,
+                            0.25, color, 1)
+
+        if len(boxes) > 0:
+            return boxes
+        else:
+            return None
```
Comment (arekmula) [inline src/inference/hand_detection.py]: always return boxes, even if its empty list

File: src/inference/hand_detection.py (hunk 7 above)
Comment (arekmula) [inline src/inference/hand_detection.py]: Add docstring about function return. I mean order of variables in each box element

File: src/inference/hand_detection.py (hunk 8 above)
Comment (arekmula) [inline src/inference/hand_detection.py]: You draw the detections on the image if "draw" is True, but you don't do anything with it. Return img_resize alongside boxes. 

So the function should always:
`return boxes, img_resize`

File: src/inference/hand_detection.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+
+if __name__ == '__main__':
```
Comment (arekmula) [inline src/inference/hand_detection.py]: Delete whole main. We don't need this

File: src/inference/hand_detection.py [hunk 10]
```diff
@@ -12,11 +9,23 @@ class HandDetector:
 
     def __init__(self):
```
Comment (arekmula) [inline src/inference/hand_detection.py]:
```
def __init__(self, image_size=416, confidence=0.2):
    self.size = image_size
    self.confidence = confidence
    self.yolo.size = int(self.size)
    self.yolo.confidence = float(self.confidence)
```

**Review by Krisisss — APPROVED** [review]
Looks good

**Review by arekmula — APPROVED** [review]

**Pull request merged**